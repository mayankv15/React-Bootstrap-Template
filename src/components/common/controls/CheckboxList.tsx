import { Checkbox, FormControlLabel, Grid } from "@mui/material";
import * as React from "react";
import SIZE from "../../../theme/font_size.json";
import COLORS from "../../../theme/colors.json";
import FONTWEIGHT from "../../../theme/font_weight.json";
import { CheckCircle, RadioButtonUnchecked } from "@mui/icons-material";
import { useTranslation } from "react-i18next";
import { makeStyles } from "@mui/styles";

const useStyles = makeStyles((theme) => ({
  gridContainer: {
    "& .MuiFormControlLabel-root": {
      display: "flex",
      flexDirection: "column",
      minWidth: 100,
      minHeight: 100,
      padding: SIZE[15],
      backgroundColor: COLORS.WHITE,
      borderRadius: SIZE[18],
      justifyContent: "center",
      margin: SIZE[10],
      maxHeight: 100,
      maxWidth: 100,
      "&.active": {
        color: theme.palette.primary.main,
        borderWidth: SIZE[1],
        borderStyle: "solid",
        borderColor: theme.palette.primary.main,
        boxShadow: "0px 2px 21px 8px rgb(69 67 67 / 39%)",
        backgroundColor: theme.palette.primary.main,
      },
      "&.active label": {
        fontWeight: FONTWEIGHT[700],
      },
      "& .MuiFormControlLabel-label .custom-label img": {
        width: 50,
      },
    },
    "& .MuiRadio-colorSecondary.Mui-checked": {
      color: theme.palette.primary.main,
    },
    "& .MuiGrid-item": {
      display: "flex",
      flexDirection: "column",
      minWidth: "20%",
      "& > label": {
        fontSize: SIZE[14],
        color: COLORS.GRAY,
      },
      "&.active label": {
        color: theme.palette.primary.main,
      },
    },
    "& .MuiGrid-item .MuiFormControlLabel-root": {
      flexGrow: 1,
      position: "relative",
      "& .MuiCheckbox-root": {
        position: "absolute",
        top: 0,
        right: 0,
      },
    },
  },
}));

interface CheckBoxListProps {
  list: any[];
  onChange?: (event: any, list: any[]) => void;
}

const CheckboxList = ({ list, onChange }: CheckBoxListProps) => {
  const classes = useStyles();
  const [inpList, setInpList] = React.useState(list);
  const { t } = useTranslation();

  const handleChange = (e, item) => {
    inpList.forEach((el) =>
      el.value === item.value ? (el.isActive = e.target.checked) : ""
    );
    setInpList([...list]);
    if (onChange) {
      onChange(e, inpList);
    }
  };

  const getDesc = (item) => {
    return (
      <div className={"custom-label"}>
        <img
          src={item.isActive ? item.inActiveIcon : item.icon}
          alt={item.title}
        />
      </div>
    );
  };
  return (
    <div className="zakat-cal-step2">
      <Grid
        container
        spacing={2}
        justifyContent="center"
        className={classes.gridContainer}
      >
        {inpList &&
          inpList.map((item, i) => (
            <Grid
              item
              className={inpList[i].isActive ? "active" : "inactive"}
              key={item.value}
              md={item.col}
              lg={item.col}
              sm={4}
            >
              <FormControlLabel
                value={item.value}
                control={
                  <Checkbox
                    icon={<RadioButtonUnchecked />}
                    checkedIcon={<CheckCircle />}
                    checked={inpList[i].isActive}
                    onChange={(e) => handleChange(e, inpList[i])}
                    value={item.value}
                  />
                }
                label={getDesc(inpList[i])}
                className={inpList[i].isActive ? "active" : "inactive"}
              />
              <label
                style={{
                  display: "flex",
                  justifyContent: "center",
                }}
              >
                {t(inpList[i].title)}
              </label>
            </Grid>
          ))}
      </Grid>
    </div>
  );
};
export default CheckboxList;
