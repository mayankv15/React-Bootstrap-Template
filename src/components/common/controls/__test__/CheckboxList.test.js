import React, { useState } from "react";
import { render, fireEvent } from "@testing-library/react";
import CheckboxList from "../CheckboxList"

let checkboxListData = [
    {
        title: 'TEST TITLE 1',
        value: '1',
        // icon: oneIcon,
        // inActiveIcon: oneInActive
    },
    {
        title: 'TEST TITLE 2',
        value: '2',
        // icon: twoIcon,
        // inActiveIcon: twoInActive
    },
    {
        title: 'TEST TITLE 3',
        value: '3',
        // icon: threeIcon,
        // inActiveIcon: threeInActive
    },
]

// const [checkList, setCheckList] = useState([...checkboxListData]);

describe("Renders CheckboxList", () => {
    it("should render CheckboxList correctly", () => {
        const setCheckList = (list) => {
            checkboxListData = list
        }
        const component = render(<CheckboxList text="Small" size="small" list={checkboxListData} onChange={(e, list) => setCheckList(list)} />);
        expect(component).toMatchSnapshot();
    });
    it("should check onchange", () => {
        const { container, getByText } = render(<CheckboxList text="Small" size="small" list={checkboxListData} onChange={(e, list) => setCheckList(list)} />);
        const element = getByText("TEST TITLE 2")
        fireEvent.click(element)
    });
});
