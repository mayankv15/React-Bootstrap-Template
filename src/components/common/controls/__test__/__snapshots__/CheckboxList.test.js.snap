// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CheckboxList should render CheckboxList correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="zakat-cal-step2"
      >
        <div
          class="MuiGrid-root makeStyles-gridContainer-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-content-xs-center"
        >
          <div
            class="MuiGrid-root inactive MuiGrid-item MuiGrid-grid-sm-4"
          >
            <label
              class="MuiFormControlLabel-root inactive"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-5"
                    data-indeterminate="false"
                    type="checkbox"
                    value="1"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                <div
                  class="custom-label"
                >
                  <img
                    alt="TEST TITLE 1"
                  />
                </div>
              </span>
            </label>
            <label
              style="display: flex; justify-content: center;"
            >
              TEST TITLE 1
            </label>
          </div>
          <div
            class="MuiGrid-root inactive MuiGrid-item MuiGrid-grid-sm-4"
          >
            <label
              class="MuiFormControlLabel-root inactive"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-5"
                    data-indeterminate="false"
                    type="checkbox"
                    value="2"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                <div
                  class="custom-label"
                >
                  <img
                    alt="TEST TITLE 2"
                  />
                </div>
              </span>
            </label>
            <label
              style="display: flex; justify-content: center;"
            >
              TEST TITLE 2
            </label>
          </div>
          <div
            class="MuiGrid-root inactive MuiGrid-item MuiGrid-grid-sm-4"
          >
            <label
              class="MuiFormControlLabel-root inactive"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-5"
                    data-indeterminate="false"
                    type="checkbox"
                    value="3"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                <div
                  class="custom-label"
                >
                  <img
                    alt="TEST TITLE 3"
                  />
                </div>
              </span>
            </label>
            <label
              style="display: flex; justify-content: center;"
            >
              TEST TITLE 3
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="zakat-cal-step2"
    >
      <div
        class="MuiGrid-root makeStyles-gridContainer-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root inactive MuiGrid-item MuiGrid-grid-sm-4"
        >
          <label
            class="MuiFormControlLabel-root inactive"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-5"
                  data-indeterminate="false"
                  type="checkbox"
                  value="1"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              <div
                class="custom-label"
              >
                <img
                  alt="TEST TITLE 1"
                />
              </div>
            </span>
          </label>
          <label
            style="display: flex; justify-content: center;"
          >
            TEST TITLE 1
          </label>
        </div>
        <div
          class="MuiGrid-root inactive MuiGrid-item MuiGrid-grid-sm-4"
        >
          <label
            class="MuiFormControlLabel-root inactive"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-5"
                  data-indeterminate="false"
                  type="checkbox"
                  value="2"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              <div
                class="custom-label"
              >
                <img
                  alt="TEST TITLE 2"
                />
              </div>
            </span>
          </label>
          <label
            style="display: flex; justify-content: center;"
          >
            TEST TITLE 2
          </label>
        </div>
        <div
          class="MuiGrid-root inactive MuiGrid-item MuiGrid-grid-sm-4"
        >
          <label
            class="MuiFormControlLabel-root inactive"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-2 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-5"
                  data-indeterminate="false"
                  type="checkbox"
                  value="3"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              <div
                class="custom-label"
              >
                <img
                  alt="TEST TITLE 3"
                />
              </div>
            </span>
          </label>
          <label
            style="display: flex; justify-content: center;"
          >
            TEST TITLE 3
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
